File: iOS/TUILiveKit/Sources/Component/AudienceList/Store/AudienceListState.swift
//
//  AudienceListState.swift
//  TUILiveKit
//
//  Created by krabyu on 2024/10/31.
//

import Foundation
import RTCRoomEngine

public class AudienceListState {
    public var roomId: String = ""
    public var ownerId: String = ""
    @Published var audienceCount: Int = 0
    @Published var audienceList: [TUIUserInfo] = []
}

let kMaxShowUserCount = 100

File: Flutter/livekit/lib/component/audience_list/state/audience_list_state.dart
import 'dart:async';

import 'package:flutter/material.dart';
import 'package:rtc_room_engine/api/room/tui_room_define.dart';

const int maxShowUserCount = 100;

class AudienceListState {
  String roomId = '';
  String ownerId = '';
  final ValueNotifier<int> audienceCount = ValueNotifier(0);
  final ValueNotifier<List<TUIUserInfo>> audienceList = ValueNotifier([]);
  final StreamController<String> roomDismissedSubject = StreamController.broadcast();

  void dispose() {
    if (!roomDismissedSubject.isClosed) {
      roomDismissedSubject.close();
    }
  }
}